sunpy/net/helio/hec.py

```diff
@@ -152,6 +152,8 @@ def search(self, *args, **kwargs):
             if isinstance(elem, a.Time):
                 qrdict['Time'] = elem
             elif isinstance(elem, ha.MaxRecords):
+                if elem.value > 20000:
+                    raise ValueError("Helio will only return a max of 20000 results.")
```

Comment:
@samaloney do want a warning or an error?

Reply:
Yea I think should raise and error as setting it any higher doesn't currently return anymore results but I don't have really strong view either way.

Reply:
Error it is.